src/util/irep_ids.h

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************\
+
+Module: util
+
+Author: Reuben Thomas, reuben.thomas@me.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_IREP_IDS_H
+#define CPROVER_UTIL_IREP_IDS_H
+
+#include "dstring.h"
+
+enum class idt:unsigned
+{
+#define IREP_ID_ONE(the_id) id_##the_id,
+#define IREP_ID_TWO(the_id, str) id_##the_id,
+
+#include "irep_ids.def"
+};
+
+#ifdef USE_DSTRING
+
+#define IREP_ID_ONE(the_id)                                                    \
+  static const dstringt ID_##the_id=                                           \
```

Comment:
Do not mark these `static` - that would entail a file-local copy in each translation unit. It might actually be worthwhile marking them extern in here, and instead generating the same (but then non-extern) declarations in a .cpp file.